src/inmanta/const.py

```diff
@@ -57,6 +57,8 @@ class ResourceState(str, Enum):
     cancelled = "cancelled"  # When a new version is pushed, in progress deploys are cancelled
     undefined = "undefined"  # The state of this resource is unknown at this moment in the orchestration process
     skipped_for_undefined = "skipped_for_undefined"  # This resource depends on an undefined resource
+    # Used to report non-compliance of reporting resource.
+    non_compliant = "non_compliant"
```

Comment:
This changes the range of expected input values for every handler. We should add this to the upgrade notes tht if you have handlers that override `deploy`, they may get a new unput value